src/background/services/bridge/handlers/avalanche_bridgeAsset.ts

```diff
@@ -220,39 +255,133 @@ export class AvalancheBridgeAsset extends DAppRequestHandler<BridgeActionParams>
     frontendTabId?: number
   ) => {
     const currentBlockchain = pendingAction?.displayData.currentBlockchain;
+
+    if (currentBlockchain === Blockchain.UNKNOWN) {
+      onError('Unsupported blockchain');
+      return;
+    }
+    this.featureFlagService.ensureFlagEnabled(FeatureGates.BRIDGE);
```

Comment:
Shouldn't we wrap this with a try-catch and call `onError` when needed so clients can get a bit more meaningful rejection?

```suggestion
    this.featureFlagService.ensureFlagEnabled(FeatureGates.BRIDGE);
```